rust/chains/hyperlane-cosmos/src/interchain_gas.rs

```diff
@@ -203,10 +205,34 @@ impl Indexer<InterchainGasPayment> for CosmosInterchainGasPaymasterIndexer {
         &self,
         range: RangeInclusive<u32>,
     ) -> ChainResult<Vec<(InterchainGasPayment, LogMeta)>> {
-        let result = self
-            .indexer
-            .get_range_event_logs(range, Self::interchain_gas_payment_parser)
-            .await?;
+        let logs_futures: Vec<_> = range
+            .map(|block_number| {
+                let self_clone = self.clone();
+                tokio::spawn(async move {
+                    let event_log = self_clone
+                        .indexer
+                        .get_logs_in_block(block_number, Self::interchain_gas_payment_parser)
+                        .await;
+                    (event_log, block_number)
+                })
+            })
+            .collect();
+
+        // TODO: this can be refactored when we rework indexing, to be part of the block-by-block indexing
+        let result = future::join_all(logs_futures)
+            .await
+            .into_iter()
+            .flatten()
+            .filter_map(|(res, block_no)| match res {
+                Ok(logs) => Some(logs),
+                Err(err) => {
+                    warn!(?err, ?block_no, "Failed to fetch logs for block");
```

Comment:
noting that we may wanna change this to not allow Errs when we do https://github.com/hyperlane-xyz/hyperlane-monorepo/issues/3264 - but for now happy to keep this as-is